WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/StatsModule.kt

```diff
@@ -457,7 +457,11 @@ class StatsModule {
         return if (trafficTabFeatureConfig.isEnabled()) {
             mapOf(
                 StatsSection.TRAFFIC to trafficUseCase,
-                StatsSection.INSIGHTS to insightsUseCase
+                StatsSection.INSIGHTS to insightsUseCase,
+                StatsSection.DAYS to dayStatsUseCase,
+                StatsSection.WEEKS to weekStatsUseCase,
+                StatsSection.MONTHS to monthStatsUseCase,
+                StatsSection.YEARS to yearStatsUseCase
```

Comment:
We do not use `StatsSection.DAYS`, `StatsSection.WEEKS`, `StatsSection.MONTHS`, and `StatsSection.YEARS` when the traffic tab is enabled. We should not add those here; otherwise, stats will fetch unused use cases from granular tabs.
To test this, you can debug `OverviewUseCase`. It should not run when the traffic tab is enabled, but you will notice that it is running.

When I pull to refresh, the `TrafficOverviewUse` should be fetched, but instead, the `OverviewUseCase` is being fetched.

Reply:
Hi @irfano, thanks for the review and feedback. I initially thought it was a mapping issue since `baseListUseCase = listUseCases[statsSectionManager.getSelectedSection()]` returns `null` in the  StatsViewModel pulltoRefresh() function.

I continued debugging today and I am noticing that both `TrafficOverviewUseCase` and `OverviewUseCase` are being created. The pull-to-refresh will sometimes work with `feature flag on` but not with `feature flag off`, and vice versa at times. Is this what you're seeing as well? I am continuing to debug why both `UseCases` are being created but open to more feedback or tips.

Reply:
Thanks for pointing out `baseListUseCase = listUseCases[statsSectionManager.getSelectedSection()]`. I debugged this on `trunk` and realized that the refresh function works fine. It stops working after switching the `stats_traffic_tab` flag. We need to kill and restart the app after changing the flag because [LIST_STATS_USE_CASES](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/StatsModule.kt#L448) is singular.
Could you please also test both cases with the flag enabled and disabled, but killing the app after switching the flag? Since debug settings are not accessible in the production environment, I believe we can close the issue without implementing any changes.

Reply:
Hi @irfano, yes both cases work as expected after killing the app. I will go ahead and close this ticket 👍